crates/stages/src/stages/finish.rs

```diff
@@ -0,0 +1,96 @@
+use crate::{
+    db::Transaction, ExecInput, ExecOutput, Stage, StageError, StageId, UnwindInput, UnwindOutput,
+};
+use reth_db::database::Database;
+
+/// The [`StageId`] of the finish stage.
+pub const FINISH: StageId = StageId("Finish");
+
+/// The finish stage.
+///
+/// This stage does nothing; it's checkpoint is used to denote the highest fully synced block,
+/// which is communicated to the P2P networking component, as well as the RPC component.
+#[derive(Default, Debug, Clone)]
+pub struct FinishStage;
+
+#[async_trait::async_trait]
+impl<DB: Database> Stage<DB> for FinishStage {
+    fn id(&self) -> StageId {
+        FINISH
+    }
+
+    async fn execute(
+        &mut self,
+        _: &mut Transaction<'_, DB>,
+        input: ExecInput,
+    ) -> Result<ExecOutput, StageError> {
+        Ok(ExecOutput { done: true, stage_progress: input.previous_stage_progress() })
```

Comment:
Does this need to call update_head somehow? Or trigger something downstream that updates the head for the p2p status msg

Reply:
Yes, that's why it's a draft